python/ray/data/tests/test_stats.py

```diff
@@ -1234,6 +1234,28 @@ def test_dataset_name():
     )
 
 
+def test_op_metrics_logging():
+    logger = DatasetLogger(
+        "ray.data._internal.execution.streaming_executor"
+    ).get_logger()
+    with patch.object(logger, "info") as mock_logger:
+        ray.data.range(100).map_batches(lambda x: x).materialize()
```

Comment:
Nit: If it's not important that we have many rows, `1` might be a better value. Writing minimally passing tests is a good practice. https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#write-minimally-passing-tests
```suggestion
        ray.data.range(1).map_batches(lambda x: x).materialize()
```

Reply:
Here I want the loop to run more than once just to check if the log is printing exactly once per operator and not every loop. I can probably reduce the size a bit